share/qtcreator/templates/wizards/sailfishos-qml-file/wizard.json

```diff
@@ -0,0 +1,81 @@
+{
+    "version": 1,
+    "supportedProjectTypes": [ ],
+    "id": "E.QmlFile",
+    "category": "Sailfish OS",
+    "trDescription": "Creates a QML file for Sailfish OS application project.",
```

Comment:
It would be a good to think about using a parameterized OS name from `Sfdk::Sdk::osVariant()`

Reply:
Do you mean for "category"?

Reply:
For all string `Sailfish OS`, `Sailfish SDK`, `Sailfish IDE` which are displayed in the interface. OS name can now be configured.

Reply:
@etaishev We checked possibility of setting the `"category"` and `"trDisplayCategory"` parameters as the wizard variables like `"%{SailfishOSCategory}"` that initialized inside C++ code using `Sfdk::Sdk::osVariant()` and it's impossible.

The `"category"` and `"trDisplayCategory"` parameters are used exactly the string that is passed instead of the variable value. So, in case you suggest user will see only "%{SailfishOSCategory}" instead of "Sailfish OS" as a category.

Reply:
Check this https://github.com/sailfishos/sailfish-qtcreator/blob/master/share/qtcreator/templates/wizards/projects/vcs/git/wizard.json#L27 - I guess the indirection with "options" is needed.

Reply:
The "options" object doesn't help too.
Such things as the "category" and "trDisplayCategory" fields are defined inside the [`IWizardFactory`](https://github.com/sailfishos/sailfish-qtcreator/blob/master/src/plugins/coreplugin/iwizardfactory.h#L45) class in the `coreplugin` Qt Creator plugin. 
The class contains following declarations for the fields methods:

```cpp
QString category() const { return m_category; }
QString displayCategory() const { return m_displayCategory; }
...
void setDisplayName(const QString &displayName) { m_displayName = displayName; }
void setCategory(const QString &category) { m_category = category; }
```

Fields are set by exactly the string that is passed ("%{SailfishOSCategory}" in our case). There is no the [`MacroExpander`](https://github.com/sailfishos/sailfish-qtcreator/blob/188bf6777d3df8d4389c687119887f18fdecc564/src/libs/utils/macroexpander.h) instance that is used inside other wizards pages to retrieve the wizard variable value, for example, inside [`ListField::initializeData()`](https://github.com/sailfishos/sailfish-qtcreator/blob/d5ef6a3b84666e082556aa69b2afb73e4886f312/src/plugins/projectexplorer/jsonwizard/jsonfieldpage.cpp#L901) method.

Reply:
We should try to improve upstream to support that then.

Reply:
So, I check this conversation as "resolved". Ok?

Reply:
Not really - please read my previous message as "You should try..." :)

Reply:
@martyone @etaishev We researched a possibility of improving upstream to support setting the wizard categry with wizard variable and it is impossible too.

Class `IWizardFactory` with methods to set up the category contains inside the `coreplugin` Qt Creator plugin. To set up new wizard variables we need the `JsonWizard` class instance and this instance should be initialized before initializing instance of `IWizardFactory` that is impossible because wizard factory are created earlier than any wizard. 

Also the `IWizardFactory` class and classes of JSON-based wizards are placed inside different plugins: `coreplugn` for `IWizardFactory` and `projectexplorer` for JSON-based wizards. The `projectexplorer` plugin already depends on `coreplugin` and we cannot make a loop of dependencies.

So, we cannot set up the wizard category with a variable. And we think it is not a problem. All standard wizard categories are set as simple translatable strings.

Reply:
"Impossible" is too strong word :) The display name and category from the json is processed here https://github.com/sailfishos/sailfish-qtcreator/blob/master/src/plugins/projectexplorer/jsonwizard/jsonwizardfactory.cpp#L602 and that is the place where additional processing could happen I think. Maybe simply by processing it with a MacroExpander instantiated right there, right for that purpose. Macros for Sailfish SDK branding could be then registered with Utils::globalMacroExpander() if they are not yet.